vision/standard_object/odlc_image_processing.py

```diff
@@ -0,0 +1,67 @@
+"""
+Preprocesses images obtained during airdrop area flyover for
+use in contour detection for the standard objects.
+"""
+
+import cv2
+import numpy as np
+
+from vision.common.constants import Image, Kernel
+
+
+def preprocess_std_odlc(image: Image, thresh_min: int, thresh_max: int) -> Image:
+    """
+    Preprocesses image for use in detecting the contours of standard objects.
+
+    Parameters
+    ----------
+    image : Image
+        image from airdrop area flyover before any processing has occured
+    thresh_min: int
+        The minimum threshold input for the Canny edge detection
+    thresh_max: int
+        The maximum threshold input for the Canny edge detection
+
+    Returns
+    -------
+    edges : Image
+        the image after preprocessing for use in contour detection/processing
+    """
+    grayscaled: Image = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)  # Convert to graycsale
+
+    blurred: Image = cv2.GaussianBlur(grayscaled, ksize=(3, 3), sigmaX=1.5)  # Blur the image
+
+    edges: Image = cv2.Canny(image=blurred, threshold1=thresh_min, threshold2=thresh_max)
+
+    # Create the kernel for the dilation
+    kernel: Kernel = np.ones((3, 3), np.uint8)
+
+    dilated: Image = cv2.dilate(edges, kernel, iterations=1)
```

Comment:
This is still a binarized image, correct? Not saying it isn't, just making sure you check that.

Reply:
So apparently it's not _really_ binarized, it just only has values of 0 or 255. The OpenCV documentation states that all nonzeros are treated as ones for `cv2.findContours()`, so it effectively performs an internal binarization no matter what we pass in.